authentication/ruby/oauth/microsoft_kiota_authentication_oauth/lib/microsoft_kiota_authentication_oauth/oauth_access_token_provider.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+require 'microsoft_kiota_abstractions'
+require 'oauth2'
+require_relative 'extensions/oauth2_ext'
+require_relative 'contexts/client_credential_context'
+require_relative 'contexts/authorization_code_context'
+require_relative 'contexts/on_behalf_of_context'
+require_relative 'contexts/oauth_context'
+
+module MicrosoftKiotaAuthenticationOAuth
+  # Access Token Provider class implementation
+  class OAuthAccessTokenProvider
+    include Concurrent::Async
+    # This is the initializer for OAuthAccessTokenProvider.
+    # :params
+    #   token_request_context: a instance of one of our token request context or a custom implementation
+    #   allowed_hosts: an array of strings, where each string is an allowed host, default is empty
+    #   scopes: an array of strings, where each string is a scope, default is empty array 
+    def initialize(token_request_context, allowed_hosts = [], scopes = [])
+      raise StandardError, 'Parameter token_request_context cannot be nil.' if token_request_context.nil?
+
+      if !@token_request_context.instance_of?(MicrosoftKiotaAuthenticationOAuth::OAuthContext)
```

Comment:
```suggestion
      if !@token_request_context.is_a?(MicrosoftKiotaAuthenticationOAuth::OAuthContext)
```
`instance_of` doesn't check for child classes. The error message could also be more specific to point devs to extending OAuthContext